### ISSUE TITLE
Generating an unmarked duplicate Disproportionation  reactions

### ISSUE THREAD

**Issue by rwest** [ReactionMechanismGenerator/RMG-Py#1823]
### Bug Description

When converting a chemkin file generated by RMG into cantera using ck2cti I got this error:
```
***********************************************************************
CanteraError thrown by Kinetics::checkDuplicates:
Undeclared duplicate reactions detected:
Reaction 2619: HNO2(571) + HNO2(699) <=> H2NO2(172) + NO2(91)
Reaction 2498: HNO2(571) + HNO2(699) <=> H2NO2(172) + NO2(91)
***********************************************************************
```

Looking at the annotated chemkin file, sure enough these reactions  are unmarked duplicates:

```
! Reaction index: Chemkin #2498; RMG #3394
! Template reaction: Disproportionation
! Estimated using an average for rate rule [Y_1centerbirad;N3H_s_Rrad]
! Euclidian distance = 0
! family: Disproportionation
HNO2(571)+HNO2(699)=NO2(91)+H2NO2(172)              2.645406e+08 1.500     -0.350   

! Reaction index: Chemkin #2619; RMG #26909
! Template reaction: Disproportionation
! Flux pairs: HNO2(699), NO2(91); HNO2(571), H2NO2(172); 
! Estimated using an average for rate rule [Y_1centerbirad;N3H_s_Rrad]
! Euclidian distance = 0
! family: Disproportionation
HNO2(699)+HNO2(571)=NO2(91)+H2NO2(172)              2.645406e+08 1.500     -0.350   
```

They even have exactly the same rate rules.
The reactants are listed in a different order, however.

### How To Reproduce

The species in question are

```
HNO2(699)
multiplicity 3
1 O u1 p2 c0 {3,S}
2 O u1 p2 c0 {3,S}
3 N u0 p1 c0 {1,S} {2,S} {4,S}
4 H u0 p0 c0 {3,S}

HNO2(571)
multiplicity 3
1 O u0 p2 c0 {2,S} {3,S}
2 O u0 p2 c0 {1,S} {4,S}
3 N u2 p1 c0 {1,S}
4 H u0 p0 c0 {2,S}

NO2(91)
multiplicity 2
1 O u0 p3 c-1 {3,S}
2 O u0 p2 c0 {3,D}
3 N u1 p0 c+1 {1,S} {2,D}

H2NO2(172)
multiplicity 2
1 O u0 p2 c0 {2,S} {3,S}
2 O u0 p2 c0 {1,S} {4,S}
3 N u1 p1 c0 {1,S} {5,S}
4 H u0 p0 c0 {2,S}
5 H u0 p0 c0 {3,S}
```

So hopefully putting the two reactants in an input file will reproduce.

### Expected Behavior
Duplicates would be eliminated, merged, or marked as DUPLICATE in the chemkin output.

### Installation Information
Describe your installation method and system information.
 - OS (include version if known): CentOS 7
 - Installation method: from source in anaconda environment
 - RMG version information:
   
   - RMG-Py: 2.4.1-511-gd33df776a
   - RMG-database: 2.4.0-37-gcc0c77f9

### Possible links
- #1624 is about when a Disproportionation and a PDep reaction end up being the same and are not marked as DUPLICATE. But in this case they are both Disproportionation.
- #1336 might be a duplicate of this issue (sorry)

**Comment by rwest** [comment]
There was one other duplicate pair in the same model (ck2cti only reports them one at a time)
```
***********************************************************************
CanteraError thrown by Kinetics::checkDuplicates:
Undeclared duplicate reactions detected:
Reaction 2624: HNO2(699) + HNOO(572) <=> H2NO2(172) + NO2(91)
Reaction 2499: HNO2(699) + HNOO(572) <=> H2NO2(172) + NO2(91)
***********************************************************************

! Reaction index: Chemkin #2624; RMG #27079
! Template reaction: Disproportionation
! Flux pairs: HNO2(699), NO2(91); HNOO(572), H2NO2(172); 
! Estimated using an average for rate rule [O_rad/NonDeO;N3H_s_Rrad]
! Euclidian distance = 0
! family: Disproportionation
HNO2(699)+HNOO(572)=NO2(91)+H2NO2(172)              2.900000e+04 2.690     -1.600 

! Reaction index: Chemkin #2499; RMG #3395
! Template reaction: Disproportionation
! Estimated using an average for rate rule [O_rad/NonDeO;N3H_s_Rrad]
! Euclidian distance = 0
! family: Disproportionation
HNOO(572)+HNO2(699)=NO2(91)+H2NO2(172)              2.900000e+04 2.690     -1.600   
```

**Comment by mliu49** [comment]
I think #1336 is related.

**Comment by rwest** [comment]
All recent examples of this have either the reactants or the products in a different order.
eg. A + B = C + D is not detected as a duplicate of A + B = D + C.

The `chemkin.mark_duplicate_reaction` function does not sort the reactant or product lists before comparing them.

After fixing this, there's a residual problem. The function REMOVES duplicate flags if it thinks they don't belong, but they might belong _for a different reason_ than the one being checked.

Attempts to fix these are in PR #1856
